lib/data/kinetic/backend.js

```diff
@@ -0,0 +1,159 @@
+import crypto from 'crypto';
+import kinetic from 'kineticlib';
+import stream from 'stream';
+import net from 'net';
+
+import { errors } from 'arsenal';
+
+import config from '../../Config';
+
+let sequence = 1;
+
+const kDrives = {
+    port: config.kinetic ? config.kinetic.port : 8123,
+    host: config.kinetic ? config.kinetic.host : 'localhost',
+};
+
+function propError(pdu) {
+    const statusCode = pdu.getStatusCode();
+    if (statusCode !== kinetic.errors.SUCCESS) {
+        if (statusCode === kinetic.errors.NOT_FOUND) {
+            return errors.ObjNotFound;
+        }
+        return pdu.getErrorMessage();
+    }
+    return undefined;
+}
+
+function newSocket(callback) {
+    const socket = new net.Socket({ allowHalfOpen: false }).pause();
+    socket.setKeepAlive(true);
+    socket.unref();
+    return socket.connect(kDrives, () => {
+        kinetic.streamToPDU(socket, (err, pdu) => {
+            if (err) {
+                return callback(err);
+            }
+            const err1 = propError(pdu);
+            return callback(err1, socket);
+        });
+    });
+}
+
+function putKinetic(socket, value, options, callback) {
+    const key = crypto.randomBytes(20);
+    const obj = Buffer.concat(value);
+    const tag = crypto.createHmac('sha1', 'asdfasdf').update(obj).digest();
+    const pdu = new kinetic.PutPDU(sequence, key, obj.length, tag, options);
+    ++sequence;
+
+    const header = pdu.read();
+    const len = header.length + obj.length;
+
+    socket.write(Buffer.concat([header, obj], len), err => {
+        if (err) {
+            return callback(err);
+        }
+        kinetic.streamToPDU(socket, (err, pdu) => {
+            const err1 = propError(pdu);
+            socket.destroy();
```

Comment:
So I investigated and it seems to come from StreamToPDU that using posix-read. So I guess the kineticlib will to have some changes. I think it could be better to have this changes in another PR.